api/tests/Unit/PoolPolicyTest.php

```diff
@@ -0,0 +1,299 @@
+        $this->assertTrue($this->requestResponderUser->can('viewAnyPublishedAdvertisement', Pool::class));
+        $this->assertTrue($this->adminUser->can('viewAnyPublishedAdvertisement', Pool::class));
+    }
+
+    /**
+     * Assert that the following can view a pool:
+     *
+     * pool operator, platform admin
+     *
+     * @return void
+     */
+    public function testView()
+    {
+        $this->assertTrue($this->poolOperatorUser->can('view', $this->teamPool));
+        $this->assertTrue($this->adminUser->can('view', $this->teamPool));
+
+        $this->assertFalse($this->guestUser->can('view', $this->teamPool));
+        $this->assertFalse($this->applicantUser->can('view', $this->teamPool));
+        $this->assertFalse($this->requestResponderUser->can('view', $this->teamPool));
+        // Pool operator cannot view other teams pools
+        $this->assertFalse($this->poolOperatorUser->can('view', $this->unOwnedPool));
+    }
+
+    /**
+     * Assert that the following can view a draft pool advertisement:
+     *
+     * pool operator, platform admin
+     *
+     * @return void
+     */
+    public function testViewDraftAdvertisement()
+    {
+        $this->teamPool->published_at = null;
+        $this->teamPool->save();
+
+        $this->assertTrue($this->poolOperatorUser->can('viewAdvertisement', $this->teamPool));
+        $this->assertTrue($this->adminUser->can('viewAdvertisement', $this->teamPool));
+
+        $this->assertFalse($this->guestUser->can('viewAdvertisement', $this->teamPool));
+        $this->assertFalse($this->applicantUser->can('viewAdvertisement', $this->teamPool));
+        $this->assertFalse($this->requestResponderUser->can('viewAdvertisement', $this->teamPool));
+
+        // Pool operator cannot view other teams draft pools
+        $this->unOwnedPool->published_at = null;
+        $this->unOwnedPool->save();
+
+        $this->assertFalse($this->poolOperatorUser->can('viewAdvertisement', $this->unOwnedPool));
+    }
+
+    /**
+     * Assert that anyone can view a published pool advertisement:
+     *
+     * @return void
+     */
+    public function testViewPublishedAdvertisement()
+    {
+        $this->teamPool->published_at = config('constants.past_date');
+        $this->teamPool->save();
+
+        $this->assertTrue($this->guestUser->can('viewAdvertisement', $this->teamPool));
+        $this->assertTrue($this->applicantUser->can('viewAdvertisement', $this->teamPool));
+        $this->assertTrue($this->poolOperatorUser->can('viewAdvertisement', $this->teamPool));
+        $this->assertTrue($this->requestResponderUser->can('viewAdvertisement', $this->teamPool));
+        $this->assertTrue($this->adminUser->can('viewAdvertisement', $this->teamPool));
+    }
+
+    /**
+     * Assert that only pool operators can create pools
+     *
+     * @return void
+     */
+    public function testCreate()
+    {
+        $this->assertTrue($this->poolOperatorUser->can('create', Pool::class));
```

Comment:
Will this pass, even with our injected args requirement? 🤔

Reply:
No this is the one test not working at this point. I can't find any way of passing some args to this.

Reply:
I remember seeing another test case like this - I'll try to find it.

Reply:
Try this:
```
public function testCreate()
    {
        $injected = [
            'team' => [
                'connect' => $this->team->id,
            ]
        ];

        $this->assertTrue($this->poolOperatorUser->can('create', [Pool::class, $injected]));

        $this->assertFalse($this->guestUser->can('create', [Pool::class, $injected]));
        $this->assertFalse($this->applicantUser->can('create', [Pool::class, $injected]));
        $this->assertFalse($this->requestResponderUser->can('create', [Pool::class, $injected]));
        $this->assertFalse($this->adminUser->can('create', [Pool::class, $injected]));
    }
```

Reply:
Thank you! I'm not sure how I never found that. Updated with [5e2dea9](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5927/commits/5e2dea9e76d7e2481ff3dbce4830684fab995b5c).